bayesian/api/user_api.py

```diff
@@ -0,0 +1,70 @@
+"""Definition of all REST API endpoints for user management."""
+import uuid
+
+from flask import Blueprint, request
+from flask.json import jsonify
+
+from fabric8a_auth.auth import login_required
+from fabric8a_auth.errors import AuthError
+
+from bayesian.exceptions import HTTPError
+from bayesian.utility import user_utils
+from bayesian.utility.user_utils import UserException
+
+user_api = Blueprint('user_api', __name__, url_prefix='/user')
+
+
+@user_api.route('/readiness')
+def readiness():
+    """Handle the /readiness REST API call."""
+    return jsonify({}), 200
+
+
+@user_api.route('/liveness')
+def liveness():
+    """Handle the /liveness REST API call."""
+    return jsonify({}), 200
+
+
+@user_api.route('/<user_id>', methods=['GET'])
+@login_required
+def get_user(user_id):
+    """Endpoint for getting user details."""
+    if not user_id:
+        raise HTTPError(400, "user id should be present")
+
+    user = user_utils.get_user(user_id)
+    return jsonify(user_id=user.user_id, snyk_api_token=user.snyk_api_token)
+
+
+@user_api.route('', methods=['POST'])
+@login_required
+def create_user():
+    """Endpoint for creating user details."""
+    user_uuid = uuid.uuid4()
+    user_utils.create_user(user_uuid, "SNYK")
```

Comment:
In case of exception while inserting user in rds, we will still be returning the uuid. Is it intended?

Reply:
No, if there is an exception it is not caught and propagated up after which the exception handler will handle it. So UUID will not be returned.

Reply:
what would be the response that will be sent back to the API caller?